apps/src/lib/client/utils.rs

```diff
@@ -591,39 +592,189 @@ pub fn default_base_dir(
     );
 }
 
+/// Derive and print all established addresses from the provided
+/// genesis txs toml file.
+pub fn derive_genesis_addresses(
```

Comment:
This command is only for testing, right? Or do we expect to be using the output in some way?